### PR TITLE
Migrating query suggestions to new OpenSearch client.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This PR is making use of #17760 to migrate query suggestions to the new OpenSearch client.

Refs #14788.

/prd Graylog2/graylog-plugin-enterprise#6679

/nocl

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/ClusterAdapterOS2.java [hunk 1]
```diff
@@ -110,16 +101,26 @@ public Set<NodeFileDescriptorStats> fileDescriptorStats() {
...
     private List<NodeResponse> nodes() {
-        final List<NodeResponse> allNodes = catApi.nodes();
-        final List<NodeResponse> nodesWithDiskStatistics = allNodes.stream().filter(NodeResponse::hasDiskStatistics).collect(Collectors.toList());
+        final var response = client.execute(c -> c.cat().nodes(builder -> builder.fullId(true)
+                .headers("id,name,role,host,ip,fileDescriptorMax,diskUsed,diskTotal,diskUsedPercent")));
+        final var allNodes = response.valueBody()
+                .stream()
+                .map(this::createNodeResponse)
+                .toList();
+        final var nodesWithDiskStatistics = allNodes.stream().filter(NodeResponse::hasDiskStatistics).toList();
```
Comment (janheise) [inline graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/ClusterAdapterOS2.java:110]: How about using a groupingBy Collector to split the stream in one step?

Comment (dennisoelkers) [reply graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/ClusterAdapterOS2.java:110]: Sure. I think that should go into a follow-up PR though.

**Review by janheise — APPROVED** [review]
LGTM

File: graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/ClusterAdapterOS2.java [hunk 2]
```diff
@@ -267,47 +296,39 @@ public JsonNode rawClusterStats() {
...
 
-    private NodeInfo createNodeInfo(JsonNode nodesJson) {
+    private NodeInfo createNodeInfo(org.opensearch.client.opensearch.nodes.info.NodeInfo nodeInfo) {
         return NodeInfo.builder()
-                .version(nodesJson.at("/version").asText())
-                .os(nodesJson.at("/os"))
-                .roles(toStream(nodesJson.at("/roles").elements()).map(JsonNode::asText).toList())
-                .jvmMemHeapMaxInBytes(nodesJson.at("/jvm/mem/heap_max_in_bytes").asLong())
+                .version(nodeInfo.version())
+                .os(nodeInfo.os())
```
Comment (mpfz0r) [inline graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/ClusterAdapterOS2.java:307]: This is a problem. 
The telemetry cluster resource and also archiving relied on this being a JsonNode.
Currently the telemetry does not contain any os info anymore:

```
    "nodes": {
      "imEvjtzgS7ytSLMpnGHmKQ": {
        "version": "2.11.0",
        "os": {},
        "jvm_mem_heap_max_in_bytes": 2147483648,
        "roles": [
          "ClusterManager",
          "Data",
          "Ingest",
          "RemoteClusterClient",
          "Search"
        ]
      },
```

Comment (mpfz0r) [reply graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/ClusterAdapterOS2.java:307]: in 5.2 this looks like
```
    "nodes": {
      "lB9tkcD_QWmJEf27s3YiCA": {
        "version": "1.3.13",
        "os": {
          "refresh_interval_in_millis": 1000,
          "name": "Linux",
          "pretty_name": "Amazon Linux 2",
          "arch": "amd64",
          "version": "5.15.0-1045-aws",
          "available_processors": 2,
          "allocated_processors": 2
        },
```

Comment (dennisoelkers) [reply graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/ClusterAdapterOS2.java:307]: Refs #18652.

**Pull request merged**